planet/src/main/webapp/js/authentication.js

```diff
@@ -86,12 +95,59 @@ async function generateUsername(displayName) {
     return username;
 }
 
-function addUserToDatabase(uid, name, phoneNumber, email, username) {
+function addUserToDatabase(uid, name, phoneNumber, email, username, emailVerified) {
+    // We need to be signed in to be able to add the user to the database
+    // even if the user hasn't verified their email yet
     database.ref('users/' + uid).set({
         name: name,
         email: email,
         phoneNumber: phoneNumber,
         username: username,
+        emailVerified: emailVerified,
+    }).then(() => {
+        // Once the user has been added, if the email is not verified, we sign out
+        if (!emailVerified) {
+            signOut();
+        }
+    });
+}
+
+function sendEmailVerification(user, displayName, email, phoneNumber, username) {
+    user.sendEmailVerification().then(function() {
+        const modalElement = document.getElementById('account-created-modal');
+        const line1 = document.createElement('p');
+        line1.style.textAlign = 'center';
+        line1.style.marginBottom = '0';
+        line1.innerText = 'Your account has been successfully created and a verification\
+        email has been sent to you.';
+        const line2 = document.createElement('p');
+        line2.style.textAlign = 'center';
+        line2.style.marginTop = '0';
+        line2.innerText = 'Please verify your email and log in again!';
+        modalElement.innerHTML = '';
+        modalElement.appendChild(line1);
+        modalElement.appendChild(line2);
+        closeModal('sign-up-modal');
+        openModal('account-created-modal');
+        addUserToDatabase(user.uid, displayName, phoneNumber, email, 
+            username, user.emailVerified);
+    })
+    .catch(function(error) {
+        const errorCode = error.code;
+        const errorMessage = error.message;
+        if (errorCode === 'auth/missing-android-pkg-name') {
+            alert('Android package name is missing');
+        } else if (errorCode === 'auth/missing-continue-uri') {
+            alert('Continue URL is missing');
```

Comment:
what is the user supposed to do about these alerts? If there's no user action that can be taken, we can show the user a more friendly generic error and log the real error

Reply:
No! The user can't do anything about them. It's probably better to just log the error on the console